### ISSUE TITLE
Support CockroachDB in Migrate

### ISSUE THREAD

**Issue by steebchen** [prisma/prisma#4702]
A tracking issue for CockroachDB

related: https://github.com/cockroachdb/appdev-issues/issues/12

List of broken tests: https://gist.github.com/pimeys/e63b5244ad58b2b01edeb8409f593d1f

**Comment by andreimc** [comment]
any updates on this one ?

**Comment by Jolg42** [comment]
Looks like this issue is a blocker for https://github.com/cockroachdb/cockroach/issues/58615

**Comment by filiplindqvist-netlight** [comment]
I think that `pg_advisory_lock` is still blocker for using migrate. Would it be possible to make prisma migrate check if advisory locks are available or not, before using them and skip locking if not supported.

**Comment by avhust** [comment]
> I think that `pg_advisory_lock` is still blocker for using migrate. Would it be possible to make prisma migrate check if advisory locks are available or not, before using them and skip locking if not supported.

Good idea!
Or maybe some sort of flag to not use pg_advisory_lock while migrate.

**Comment by alexanderniebuhr** [comment]
hey everyone 👋
I have looked at some issues for prisma & cockroach ..

Some people hinted that it might work patial, I can confirm that.
Now I am trying to change my db with `prisma migrate` to my defined schema

Currently I do get the following error, I am not sute if it should work or not, I have not found any other issue regarding this issue. My question now, **is this something I could fix with an setting or a workaround or do I have to wait for more stable support of cockroach in prisma?**

```
❯ npx prisma migrate dev --name init
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "hocus", schema "public" at "****.****.io:26257"

Error: Database error
Error querying the database: db error: ERROR: unknown function: pg_get_functiondef()
   0: sql_migration_connector::flavour::postgres::sql_schema_from_migration_history
             at migration-engine\connectors\sql-migration-connector\src\flavour\postgres.rs:278
   1: migration_core::api::DevDiagnostic
             at migration-engine\core\src\api.rs:93
```
```
❯ npx prisma db push
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "hocus", schema "public" at "****.****:26257"
Error: Database error
Error querying the database: db error: ERROR: unknown function: pg_get_functiondef()
   0: migration_core::api::SchemaPush
             at migration-engine\core\src\api.rs:165
```

**Comment by janpio** [comment]
No, this is one of the many things that work in standard PostgreSQL but not in CockroachDB. Even when you get rid of all of these, which is possible, you are left with differences in how data is represented which will Migrate try to migrate and update your schema each time you run it in areas where you did not even make changes. This will require deeper changes in Prisma, which is why this is hard and will take some time.

**Comment by uncvrd** [comment]
I'd love to hear if there's renewed interest in compatibility due to the release of CockroachDB Serverless? From what I gather, this is more on the CockroachDB team to pick up on their side?

**Comment by dborstelmann** [comment]
I would love to use Prisma Migrate with CockroachDB Serverless.  Hosting Aurora Serverless is expensive/time consuming so if I could get migrations to work I would switch to CockroachDB

**Comment by janpio** [comment]
Related issue on Cockroach side: https://github.com/cockroachdb/cockroach/issues/73137

**Comment by floelhoeffel** [comment]
Released in preview with `3.11.0`.

For feedback please head to https://github.com/prisma/prisma/issues/11542
